### ISSUE TITLE
Configuration with nested objects

### ISSUE THREAD

**Issue by WixoLeo** [chaostoolkit/chaostoolkit-lib#195]
**Fantastic day everyone!**

Currently it's not possible to have a configuration with nested objects inside `experiment.json`.
Example:

```json
"configuration": {
    "servers": {
      "north": {
        "ip": "123.123.123.123",
        "port": "1234",
        "id": "f43e345345",
        "dashboard_id": "234234324",
        ...
      },
      "east": {
        "ip": "123.123.123.124",
        "port": "1235",
        "id": "f43e345346",
        "dashboard_id": "234234327",
        ...
      },
      ...
    }
  },
```

Now imagine that each server has many properties, and perhaps some properties are objects as well.

So in order to solve this with the current version, we have to do one of these solutions:

1. In load the separate config in the controls (perhaps configure control) and somehow save it as a global object so everything would be able to access it.
2. Have each property be named like this `"servers_east_ip"`, `"servers_east_port"` which is really not an option because these objects can go deep.
Also it's problematic because we want to select a random server object and perform an experiment on it (it's just an example).

Another thing to note, why we wouldn't want to use the 1st solution is because we want to inject this configuration when we run the experiment.

In our CLI script we have an option to run a sandbox or prod environment. Each one has it's own configuration file that we pass to the `chaos run` command using the `--var-file`.
Therefore it would be much easier if it worked with the deep objects out of the box.

What do you think?

**Comment by Lawouach** [comment]
So the pb is not in the loading (which does the right thing) but the subsitution does not keep the typing of the source variable and convert to string.

**Comment by Lawouach** [comment]
@WixoLeo this should be fixed in the master branch. I'll let you try and confirm before releasing.

**Comment by WixoLeo** [comment]
> @WixoLeo this should be fixed in the master branch. I'll let you try and confirm before releasing.

It's.... It's..... Beautiful..
Works fantastically! This saves soooo much time, you wont believe! Thank you!

**Comment by Lawouach** [comment]
Awesome to hear. I'll release now then. You'll have to update chaostoolkit-lib only :)

**Comment by Lawouach** [comment]
Released with 1.16.0
